pkg/daemon/rpm-ostree.go

```diff
@@ -214,12 +214,11 @@ func (r *RpmOstreeClient) PullAndRebase(container string, keep bool) (imgid stri
 		imgid = container
 	}
 
-	// Clean up a previous container
-	podmanRemove(pivottypes.PivotName)
+	containerName := pivottypes.PivotNamePrefix+string(uuid.NewUUID())
```

Comment:
This will have the potential side effect of "leaking" these containers if we fail and retry though.

Possibly a stronger fix here is to use `podman create --annotation=org.openshift.machineconfigoperator.pivot=true`, and then we can filter for containers with that annotation, remove them?

That said, this seems like a raw podman incompatibility across upgrades.  We may need to use a gigantic hammer there like `crio-wipe.service` is doing.

You don't have to solve all of that in this PR, but how about adding the annotation at least while we're here, so a future patch could clean things up more precisely?

Reply:
> Possibly a stronger fix here is to use `podman create --annotation=org.openshift.machineconfigoperator.pivot=true`, and then we can filter for containers with that annotation, remove them?

I can do that, prefixing has the same effect when filtering in my mind (+ grep) but annotation looks cooler

> You don't have to solve all of that in this PR, but how about adding the annotation at least while we're here, so a future patch could clean things up more precisely?

Yes and Yes, annotation sounds good to me

Reply:
adding a defer as well, so if it's erroring out, we'll purge it, let's see

Reply:
should be fixed now